crates/sui-transaction-builder/src/lib.rs

```diff
@@ -97,6 +97,79 @@ impl TransactionBuilder {
         }
     }
 
+    /// Construct the transaction data for a dry run (uses min gas budget)
+    pub async fn tx_data_for_dry_run(
+        &self,
+        sender: SuiAddress,
+        kind: TransactionKind,
+        gas_budget: u64,
+        gas_price: u64,
+        gas_payment: Option<Vec<ObjectID>>,
+        gas_sponsor: Option<SuiAddress>,
+    ) -> TransactionData {
+        let gas_payment = self
+            .input_refs(gas_payment.unwrap_or_default().as_ref())
+            .await
+            .unwrap_or_else(|_| vec![]);
+        let gas_sponsor = gas_sponsor.unwrap_or(sender);
+        TransactionData::new_with_gas_coins_allow_sponsor(
+            kind,
+            sender,
+            gas_payment,
+            gas_budget,
+            gas_price,
+            gas_sponsor,
+        )
+    }
+
+    /// Construct the transaction data from a transaction kind and a few needed
+    /// Use the `input_coins` for pay_sui
```

Comment:
This comment doesn't make a whole lot of sense to me.

Reply:
Oops, something went wrong here / forgot to fix it. Thanks!